apps/nextjs/src/components/AppComponents/download/DownloadAllButton.tsx

```diff
@@ -0,0 +1,298 @@
+import { z } from "zod";
+
+import { getLessonTrackingProps } from "@/lib/analytics/helpers";
+import useAnalytics from "@/lib/analytics/useAnalytics";
+import { ResourceFileTypeValueType } from "@/lib/avo/Avo";
+import { trpc } from "@/utils/trpc";
+
+import {
+  ExportsType,
+  getExportsConfig,
+} from "../../ExportsDialogs/exports.helpers";
+import { Icon } from "../../Icon";
+import LoadingWheel from "../../LoadingWheel";
+import LessonIcon from "../../SVGParts/LessonIcon";
+import QuizIcon from "../../SVGParts/QuizIcon";
+import SlidesIcon from "../../SVGParts/SlidesIcon";
+
+const allexportLinksObject = z.object({
+  lessonSlides: z.string(),
+  lessonPlan: z.string(),
+  worksheet: z.string(),
+  additionalMaterials: z.string(),
+  starterQuiz: z.string(),
+  exitQuiz: z.string(),
+});
+
+type DownloadAllButtonProps = {
+  onClick: () => void;
+  lesson: LooseLessonPlan;
+  title: string;
+  subTitle: string;
+  downloadAvailable: boolean;
+  downloadLoading: boolean;
+  data:
+    | {
+        lessonSlides: string | undefined;
+        lessonPlan: string | undefined;
+        worksheet: string | undefined;
+        additionalMaterials: string | undefined;
+        starterQuiz: string | undefined;
+        exitQuiz: string | undefined;
+      }
+    | {
+        error: unknown;
+        message: string;
+      }
+    | undefined;
+  exportsType: ExportsType;
+  "data-testid"?: string;
+};
+
+type zipDownloadStatus = "idle" | "loading" | "complete" | "error";
+
+export const DownloadAllButton = ({
+  onClick,
+  lesson,
+  title,
+  subTitle,
+  downloadAvailable,
+  downloadLoading,
+  data,
+  exportsType,
+  "data-testid": dataTestId,
+}: Readonly<DownloadAllButtonProps>) => {
+  const link = data && "link" in data ? data.link : "";
+  const hasError = data && "message" in data;
+  const errorMessage = data && "message" in data ? data.message : "";
+  const { track } = useAnalytics();
+  const [zipStatus, setZipStatus] = useState<zipDownloadStatus>("idle");
+  const { icon, ext, analyticsResourceType } = getExportsConfig(exportsType);
+
+  const { isSuccess, isError, mutateAsync, isLoading } =
+    trpc.exports.sendUserAllAssetsEmail.useMutation();
+
+  const { mutateAsync: zipStatusMutateAsync } =
+    trpc.exports.checkDownloadAllStatus.useMutation();
+
+  if (data) {
+    const fileIdsAndFormats:
+      | ({ fileId: string | undefined; formats: string[] } | undefined)[]
+      | undefined = data
+      ? Object.entries(data).map(([key, value]) => {
+          if (typeof value === "string") {
+            return {
+              fileId: value?.split("/edit")[0]?.split("/d/")?.[1],
+              formats:
+                key === "lessonSlides" ? ["pptx", "pdf"] : ["docx", "pdf"],
+            };
+          }
+        })
+      : undefined;
+
+    const taskId = `download-all-${JSON.stringify(fileIdsAndFormats)}`;
+
+    function trackDownload(resourceFileType: ResourceFileTypeValueType) {
+      track.lessonPlanResourcesDownloaded({
```

Comment:
Maybe double check with @emmalanches about this?

Reply:
Have checked in and we are adding a new download type